### PR TITLE
All tokens page

### PR DESCRIPTION
**Pull request by jsidorenko**

### CODE REVIEW

File: src/components/Pages/Assets/Tokens/NativeToken.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { BN, formatBalance } from '@polkadot/util';
+import { memo } from 'react';
+import styled from 'styled-components';
+
+import { NativeTokenMetadata } from '@helpers/interfaces';
+
+interface Props {
```
Comment (VadimSaveljev) [inline src/components/Pages/Assets/Tokens/NativeToken.tsx]: NativeTokenProps

File: src/components/Pages/Assets/Tokens/NativeToken.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+`;
+
+const SBalance = styled.span`
+  text-align: right;
+`;
+
+const Token = ({ metadata, balance }: Props) => {
```
Comment (VadimSaveljev) [inline src/components/Pages/Assets/Tokens/NativeToken.tsx]: NativeToken

File: src/components/Pages/Assets/Tokens/Tokens.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  nativeMetadata,
+  tokensBalances,
+  tokensMetadata,
+}: TokensProps) => {
+  useEffect(() => {
+    getNativeBalance();
+    getNativeMetadata();
+    getTokensBalances();
+    getTokensMetadata();
+  }, [getNativeBalance, getNativeMetadata, getTokensBalances, getTokensMetadata]);
```
Comment (SaltyCucumber) [inline src/components/Pages/Assets/Tokens/Tokens.tsx:35]: this useEffect scares me :D
Not sure, but I think every method here is called 4 times each

File: src/components/Pages/Assets/Tokens/TokensView.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      <Tokens
+        getNativeBalance={getNativeBalance}
+        getNativeMetadata={getNativeMetadata}
+        getTokensBalances={getTokensBalances}
+        getTokensMetadata={getTokensMetadata}
+        nativeBalance={nativeBalance}
+        nativeMetadata={nativeMetadata}
+        tokensBalances={tokensBalances}
+        tokensMetadata={tokensMetadata}
+      />
```
Comment (VadimSaveljev) [inline src/components/Pages/Assets/Tokens/TokensView.tsx:31]: is there a reason to not use useAssets inside Tokens?
A good reason would be if some state from useAssets is reused in two or more components here, but I only see one.

File: src/hooks/useAssets.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  const { api, activeAccount } = useAccounts();
+  const [tokensMetadata, setTokensMetadata] = useState<TokenMetadata[] | null>(null);
+  const [tokensBalances, setTokensBalances] = useState<TokenBalance[] | null>(null);
+  const [nativeBalance, setNativeBalance] = useState<BN | null>(null);
+  const [nativeMetadata, setNativeMetadata] = useState<NativeTokenMetadata | null>(null);
+
+  const getTokenIds = useCallback(async () => {
+    if (api) {
+      const results: StorageKey<[AssetId]>[] = await api.query.assets.asset.keys();
+      return results.map(({ args: [id] }) => id);
```
Comment (VadimSaveljev) [inline src/hooks/useAssets.ts:25]: when I was getting ids for collections and nfts they were provided in an unsorted manner when using .keys
does ordering matter here?

Comment (jsidorenko) [reply src/hooks/useAssets.ts:25]: The order doesn't matter here. The only place where the order matters is the metadata fetching method

**Review by VadimSaveljev — APPROVED** [review]

**Pull request merged**